specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,561 @@
+Alternatively, to use delegation of trust, anchors must be explicitly defined in the relevant "trust-anchor" VOA directory (see [purpose]).
+
+#### Hierarchical trust
+
+(e.g. X.509, SSH+CA)
+
+Other systems have an inherent assumption of hierarchical delegation, formalized by cryptographic certifications.
+In such systems, globally or locally accepted _trust anchors_ are used as starting points in signature validation.
+Certificates and the delegation relations between them form a tree structure.
+Such hierarchical key structures are often referred to as [public key infrastructure] (PKI).
+
+This model requires explicitly defined _trust anchors_ in VOA.
+
+### Typical distribution format of verifers / short vs. long-lived keys
+
+VOA's goal is to provide a uniform representation of verifiers, to the degree that this is possible with differing technologies.
+To achieve this, verifiers are stored in a central file hierarchy that can be used with different technologies.
+
+Some concerns may be handled differently in supported technologies:
+
+- The separation between _trust anchors_ and _artifact verifiers_ may not be reasonable to model in separated directories, because the native representation of a given technology is to store both in a shared file, as a type of "certificate chain".
+- By default, VOA suggests that each verifier is stored in an individual file (that ideally reflects a unique identifier of that verifier in its filename).
+  However, in some technologies, separating verifiers into individual files may be uncommon or unreasonable.
+  In these technologies, bundles of verifiers may be stored in a shared file (this can apply equally to _artifact verifiers_ and _trust anchors_).
+- As an edge case, in some technologies, _artifact verifiers_ may be extremely short lived, so it may not be reasonable to store them in VOA.
+  In these cases, only a set of _trust anchors_ may be stored in VOA, while the _artifact verifiers_ are distributed with the signed artifacts.
+
+## File Hierarchy
+
+The _File Hierarchy for the Verification of OS Artifacts (VOA)_ is maintained as a directory structure on a system.
+It contains cryptographic public-key material for one or more technologies.
+
+The VOA hierarchy organizes _signature verifiers_ by [os], [purpose], [context] and [technology], and stores them in a directory structure of the form `$os/$purpose/$context/$technology/`.
+
+Vendors provide a set of _signature verifier_ files for their OS, in the VOA hierarchy format.
+
+Users of a VOA hierarchy (such as package installation software) can pick the relevant set of _signature verifiers_ for the verification of a specific artifact based on their location in the hierarchy.
+
+### Load paths
+
+VOA defines a list of _load paths_ for system mode and user mode, from which a hierarchy of _signature verifiers_ can be retrieved.
+
+Load paths when in system mode:
+
+- `/etc/voa/`
+- `/run/voa/`
+- `/usr/local/share/voa/`
+- `/usr/share/voa/`
```

Comment:
Will all files be in one directory? Take a look at `distribution-gpg-keys` package, where there are thousands of keys. If all of them are in one directory, it would be a pain.

Reply:
No, the signature verifiers are not all in one directory.

See e.g. https://github.com/uapi-group/specifications/pull/134/files#diff-c79d1da1ef2fcfffc28ac7308505535eac0942d086d54a990553374ac81fed00R98 where the hierarchy is first mentioned and from where the various subdirectory layers can be accessed.